pkg/kube/volume/volume.go

```diff
@@ -131,6 +133,10 @@ func CreatePVCFromSnapshot(ctx context.Context, args *CreatePVCFromSnapshotArgs)
 	}
 	snapshotKind := "VolumeSnapshot"
 	snapshotAPIGroup := "snapshot.storage.k8s.io"
```

Comment:
Let's add a comment here saying since version is not specified here, version is figured out automatically which causes the issues. And that's the reason, we should check if args as groupVersion (with version specified explicitly) and if yes use that.